namesys/ipns_validate_test.go

```diff
@@ -136,9 +137,9 @@ func TestResolverValidation(t *testing.T) {
 	}
 
 	// Record should fail validation because entry is expired
-	_, err = resolver.resolveOnce(ctx, id.Pretty())
-	if err != ErrExpiredRecord {
-		t.Fatal("ValidateIpnsRecord should have returned ErrExpiredRecord")
+	_, err = resolver.resolveOnce(ctx, id.Pretty(), opts.DefaultResolveOpts())
+	if err == nil {
```

Comment:
Why arent we checking for a specific error anymore?

Reply:
The `resolveOnce()` [method used to call](https://github.com/ipfs/go-ipfs/pull/4733/files#diff-36d8e88d65c9668b43e0c41d99084a99L154) `routing.GetValue()` but [now we call](https://github.com/ipfs/go-ipfs/pull/4733/files#diff-36d8e88d65c9668b43e0c41d99084a99R192) `routing.GetValues()` directly.
The mock `ValueStore` implementation's `GetValue()` method would get a single record and call `VerifyRecord()` on it, which would return `ErrExpiredRecord`/`ErrSignature` etc. We now call the mock `ValueStore` implementation's [GetValues()](https://github.com/ipfs/go-ipfs/pull/4733/files#diff-a822f7067375edcc1e809bf61ae43c6dL265) method, which gets multiple records, calls `VerifyRecord()` on each one individually, and adds the record to the list of valid records if there's no error. If there are no valid records it returns `routing.ErrNotFound`.

Reply:
I looked into whether I can check for `routing.ErrNotFound` but the go-ipfs-routing mock returns `datastore.ErrNotFound` instead of `routing.ErrNotFound` when it doesn't find a key:
https://github.com/ipfs/go-ipfs-routing/blob/master/mock/centralized_client.go#L46

Reply:
Alright, that seems fine. Thanks for the response!